source/TimePicker.js

```diff
@@ -628,11 +670,11 @@
 		*/
 		setChildPickers: function (inOld) {
 			if (this.value) {
-				var hour = this.value.getHours();
-				this.$.hour.setValue(hour);
-				this.$.minute.setValue(this.value.getMinutes());
-				if (this.meridiemEnable === true) {
-					this.$.meridiem.setValue(hour > 11 ? 1 : 0);
+				var values = this.calcPickerValues();
+				this.$.hour.setValue(values.hour);
```

Comment:
Let's use the generic setter for this and the following lines i.e. `this.$.hour.set('value', values.hour);`